source/features/render-semantic-types.tsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import {$$} from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '../feature-manager.js';
+
+function isSemanticCommitTitleElement(commitTitle: string): boolean {
+	// This may not be fully accurate.
+	// Tests, if the title string
+	// starts with a word,
+	// followed by an anything in parentheses,
+	// followed by an optional exclamation mark,
+	// followed by a colon,
+	// followed by anything.
+	return /^\w*(?:\(.*\))?!?:.*/.test(commitTitle);
+}
+
+const defaultCommitTypes = new Set([
+	'feat',
+	'fix',
+	'chore',
+	'docs',
+	'build',
+	'refactor',
+	'test',
+	'ci',
+	'perf',
+]);
+
+function isSemanticCommitTitleTypeDefault(semanticCommitTitle: string): boolean {
+	// This may not be fully accurate.
+	// Gets the first word from the title.
+	const type = /^(\w*)/.exec(semanticCommitTitle)?.pop();
+
+	if (!type) {
+		return false;
+	}
+
+	return defaultCommitTypes.has(type);
+}
+
+function extractSemanticCommitTitleTypeAndScope(semanticCommitTitle: string): [string, string?] {
+	// Gets the first word from the title.
+	const type = /^(\w*)/.exec(semanticCommitTitle)!.pop()!;
+
+	// Gets the word in parentheses from the title.
+	const scope = /\((\w*)\)/.exec(semanticCommitTitle)?.pop();
+
+	return [type, scope];
+}
+
+function removeSemanticCommitTitleAndScopeFromTitleElement(semanticCommitTitleElement: HTMLElement): void {
+	const children = semanticCommitTitleElement.childNodes;
+
+	// Remove all children until the child with a colon is found.
+	// Then remove the colon and everything before it.
+	for (const child of children) {
+		if (child.textContent.includes(':')) {
+			child.textContent = child.textContent.split(':').slice(1).join(':');
+			break;
+		}
+
+		child.remove();
+	}
+}
+
+const defaultCommitTypesMapping = new Map([
+	[
+		'feat',
+		{
+			label: 'Feature',
+			colorCssRules: {
+				'--label-r': 14,
+				'--label-g': 138,
+				'--label-b': 22,
+				'--label-h': 123,
+				'--label-s': 81,
+				'--label-l': 29,
+			},
+		},
+	],
```

Comment:
CSS goes in CSS files. This shortens this file considerably. Each class can be named like `.rgh-semantic-release-feat`